src/SoftDeleteable/Query/TreeWalker/SoftDeleteableWalker.php

```diff
@@ -104,18 +106,38 @@ protected function doGetExecutorWithCompat($statement): AbstractSqlExecutor
     {
         switch (true) {
             case $statement instanceof DeleteStatement:
-                assert(class_exists($statement->deleteClause->abstractSchemaName));
-
-                $primaryClass = $this->getEntityManager()->getClassMetadata($statement->deleteClause->abstractSchemaName);
+                return $this->createDeleteStatementExecutor($statement);
+            default:
+                throw new UnexpectedValueException('SoftDeleteable walker should be used only on delete statement');
```

Comment:
I guess we can remove the `switch` statement and use a condition with an early return. WDYT?

BTW, we should add a `@phpstan-assert` annotation for this check.

```php
@phpstan-assert DeleteStatement $statement
```

Reply:
I don't think it can be correctly asserted the statement is always an instance of `DeleteStatement` since the caller here is always in the ORM and can be any of the statement classes.

As for the early return, TBH, I've gone back and forth on how to handle this since originally writing this PR because the translation walker was changed from throwing like this walker does now to just calling the parent after https://github.com/doctrine-extensions/DoctrineExtensions/issues/2013 and I'm not totally sure if keeping the throw is right or if it should be changed and the ORM allowed to do its own thing.

Reply:
IIUC, the [`@phpstan-assert`](https://phpstan.org/writing-php-code/narrowing-types#custom-type-checking-functions-and-methods) annotation should work by telling the callers that the argument will be accepted by the method, but making explicit that the method has an assertion (the `instanceof` check with the thrown exception) about the specified type.

Reply:
OK, I've just updated this all.  Upstream uses match statements so I just downgraded to switch but that also works better when the API is fully typed.